### DISCUSSION TITLE
Error while starting zigbee-herdsman

### DISCUSSION THREAD

**Discussion by madsus1** [Koenkk/zigbee2mqtt#14328] [Devices]
Hi,

I have a problem, that I hope someone can help me with.

I'm on a RPI3 and is running Z2M. I just upgraded from 1.27.0 to 1.28.0. It's after this the problem occure - I also upgraded the Raspberry Pi OS.

When I'm starting Z2M with the command "sudo systemctl start zigbee2mqtt" I get the error "Error while starting zigbee-herdsman".

When I'm starting Z2M with the command "cd /opt/zigbee2mqtt" and "npm start" there is no problem and it's starting without any hickups.

Anyone with some good suggestions? :-)

Thanks.

Br. Mads

**Comment by Koenkk** [comment]
- What adapter are you using?
- Can you provide the herdsman debug log of this?

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Reply by madsus1** [reply]
I'm using a Sonoff ZBDongel-P with the latest firmware 20220219.

I tried to log the Herdsman debug log. I'm not an expert in this regard - but I'm trying.

I have done the following.

cd /opt/zigbee2mqtt
DEBUG=zigbee-herdsman* npm start

I have found the STDOUT file, but I can't see anything from the Z2M. Hope for some advice. :-)

Thanks.

**Reply by Koenkk** [reply]
You should see the output on your screen if you run with `DEBUG=zigbee-herdsman* npm start`

**Reply by madsus1** [reply]
Okay. :-) 

I have tried to get the output from the screen - hope you can use it? If not then I can try again. See attached file.
[Logfile.txt](https://github.com/Koenkk/zigbee2mqtt/files/9710218/Logfile.txt)

**Reply by Koenkk** [reply]
I see now that you only have problems when starting as a service. Try chagning the user to `root` in  /etc/systemd/system/zigbee2mqtt.service

**Reply by madsus1** [reply]
Thanks a lot! 

I have tried to change the user to root,, but the problem is still the same. I have made a new log. Maybe you can see something in that. 


[log_05_10_2022.txt](https://github.com/Koenkk/zigbee2mqtt/files/9717169/log_05_10_2022.txt)

**Reply by Koenkk** [reply]
Can yo provide me the debug log when starting through the service? After starting the service grab the latest logs from `/opt/zigbee2mqtt/data/log`

**Reply by madsus1** [reply]
Of course - here you go. 

debug 2022-10-05 20:45:25: Loaded state from file /opt/zigbee2mqtt/data/state.json
info  2022-10-05 20:45:25: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-10-05.20-45-25' filename: log.txt
debug 2022-10-05 20:45:25: Removing old log directory '/opt/zigbee2mqtt/data/log/2022-10-05.20-43-06'
info  2022-10-05 20:45:25: Starting Zigbee2MQTT version 1.28.0 (commit #03ba647)
info  2022-10-05 20:45:25: Starting zigbee-herdsman (0.14.62)
debug 2022-10-05 20:45:25: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","databasePath":"/opt/zigbee2mqtt/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":40000},"serialPort":{"path":"/dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_2edda9bf4a2aec11ac96cc6540c9ce8d-if00-port0"}}'
error 2022-10-05 20:45:25: Error while starting zigbee-herdsman
error 2022-10-05 20:45:25: Failed to start zigbee
error 2022-10-05 20:45:25: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2022-10-05 20:45:25: Exiting...
error 2022-10-05 20:45:25: Error: The module '/opt/zigbee2mqtt/node_modules/@serialport/bindings/build/Release/bindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 108. This version of Node.js requires
NODE_MODULE_VERSION 93. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1210:18)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at bindings (/opt/zigbee2mqtt/node_modules/bindings/bindings.js:112:48)
    at Object.<anonymous> (/opt/zigbee2mqtt/node_modules/@serialport/bindings/lib/linux.js:2:36)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)

**Reply by Koenkk** [reply]
In the service file, update the path of npm (e.g. `ExecStart=/usr/bin/npm start`) with the output of `which npm`

**Comment by madsus1** [comment]
OK. 

Sorry, but the problem is still there.

I did run the which npm and the output was "ExecStart=/home/pi/.config/nvm/versions/node/v18.10.0/bin/npm". I did add a "start" in the end - don't know if that is right! 

I updated it with "ExecStart=/home/pi/.config/nvm/versions/node/v18.10.0/bin/npm start"

**The zigbee2mqtt.service file look like this.**

[Unit]
Description=zigbee2mqtt
After=network.target

[Service]
ExecStart=/home/pi/.config/nvm/versions/node/v18.10.0/bin/npm start
WorkingDirectory=/opt/zigbee2mqtt
StandardOutput=null
# Or use StandardOutput=null if you don't want Zigbee2MQTT messages filling syslog, for more options see systemd.exec(5)
StandardError=inherit
Restart=always
User=root

[Install]
WantedBy=multi-user.target

**The latest log is like below.**

debug 2022-10-05 21:21:32: Loaded state from file /opt/zigbee2mqtt/data/state.json
info  2022-10-05 21:21:32: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-10-05.21-21-32' filename: log.txt
debug 2022-10-05 21:21:32: Removing old log directory '/opt/zigbee2mqtt/data/log/2022-10-05.20-55-16'
info  2022-10-05 21:21:32: Starting Zigbee2MQTT version 1.28.0 (commit #03ba647)
info  2022-10-05 21:21:32: Starting zigbee-herdsman (0.14.62)
debug 2022-10-05 21:21:32: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","databasePath":"/opt/zigbee2mqtt/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":40000},"serialPort":{"path":"/dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_2edda9bf4a2aec11ac96cc6540c9ce8d-if00-port0"}}'
error 2022-10-05 21:21:32: Error while starting zigbee-herdsman
error 2022-10-05 21:21:32: Failed to start zigbee
error 2022-10-05 21:21:32: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2022-10-05 21:21:32: Exiting...
error 2022-10-05 21:21:32: Error: The module '/opt/zigbee2mqtt/node_modules/@serialport/bindings/build/Release/bindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 108. This version of Node.js requires
NODE_MODULE_VERSION 93. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1210:18)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at bindings (/opt/zigbee2mqtt/node_modules/bindings/bindings.js:112:48)
    at Object.<anonymous> (/opt/zigbee2mqtt/node_modules/@serialport/bindings/lib/linux.js:2:36)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)

**Reply by Koenkk** [reply]
In `/opt/zigbee`, can you execute `rm -rf node_modules && /home/pi/.config/nvm/versions/node/v18.10.0/bin/npm ci`

**Comment by madsus1** [comment]
Thanks a lot for the help. I did reinstall the full system and it's up and running. :-)

Thank you for the outstanding support and software - I'll give a cup of coffee (or five) - you deserved it!

Br. Mads

**Comment by robinfiddelaers** [comment]
I'm running to a bit the same problem here. 
Tried the solutions given here ... but no success

first I tried the 'regular' HA-addon (see other post)
Now I tried the install-procedure given on https://www.zigbee2mqtt.io/guide/installation/01_linux.html#determine-location-of-the-adapter-and-checking-user-permissions

At the moment I should run 'npm ci' I needed to run it as root
same for the nano ... config-yaml

sudo journalctl -u zigbee2mqtt.service  brings me the next message 

okt 15 09:44:37 raspberrypi systemd[1]: Started zigbee2mqtt.
okt 15 09:44:40 raspberrypi npm[7710]: > zigbee2mqtt@1.28.0 start
okt 15 09:44:40 raspberrypi npm[7710]: > node index.js
okt 15 09:44:41 raspberrypi npm[7722]: node:internal/modules/cjs/loader:959
okt 15 09:44:41 raspberrypi npm[7722]:   throw err;
okt 15 09:44:41 raspberrypi npm[7722]:   ^
okt 15 09:44:41 raspberrypi npm[7722]: Error: Cannot find module 'core-js/features/object/from-entries'
okt 15 09:44:41 raspberrypi npm[7722]: Require stack:
okt 15 09:44:41 raspberrypi npm[7722]: - /opt/zigbee2mqtt/index.js
okt 15 09:44:41 raspberrypi npm[7722]:     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
okt 15 09:44:41 raspberrypi npm[7722]:     at Function.Module._load (node:internal/modules/cjs/loader:804:27)
okt 15 09:44:41 raspberrypi npm[7722]:     at Module.require (node:internal/modules/cjs/loader:1028:19)
okt 15 09:44:41 raspberrypi npm[7722]:     at require (node:internal/modules/cjs/helpers:102:18)
okt 15 09:44:41 raspberrypi npm[7722]:     at Object.<anonymous> (/opt/zigbee2mqtt/index.js:1:1)
okt 15 09:44:41 raspberrypi npm[7722]:     at Module._compile (node:internal/modules/cjs/loader:1126:14)
okt 15 09:44:41 raspberrypi npm[7722]:     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
okt 15 09:44:41 raspberrypi npm[7722]:     at Module.load (node:internal/modules/cjs/loader:1004:32)
okt 15 09:44:41 raspberrypi npm[7722]:     at Function.Module._load (node:internal/modules/cjs/loader:839:12)
okt 15 09:44:41 raspberrypi npm[7722]:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
okt 15 09:44:41 raspberrypi npm[7722]:   code: 'MODULE_NOT_FOUND',
okt 15 09:44:41 raspberrypi npm[7722]:   requireStack: [ '/opt/zigbee2mqtt/index.js' ]
okt 15 09:44:41 raspberrypi npm[7722]: }
okt 15 09:44:41 raspberrypi systemd[1]: zigbee2mqtt.service: Main process exited, code=exited, status=1/FAILURE
okt 15 09:44:41 raspberrypi systemd[1]: zigbee2mqtt.service: Failed with result 'exit-code'.
okt 15 09:44:41 raspberrypi systemd[1]: zigbee2mqtt.service: Consumed 3.610s CPU time.

I noticed that in the installation user pi is mentioned. Obviously, I replaced it with my username

I would be glad to understand what is going wrong ...
thx in advance for insights !

scOOb

(Before post-edit :my bad ... forgot to edit the pi user in service file ... This is the correct log.  But still no luck running it)

**Reply by robinfiddelaers** [reply]
Wow ... problem solved ... Low voltage on Pi ... mèh :-/
Deleted, re-installed, tested and ... up and running
